src/App.js

```diff
@@ -1,16 +1,52 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+
+  const countLiked = () => {
+    let count = 0;
+    for (let message of messages) {
+      if (message.liked === true) {
+        count += 1;
+      }
+    }
+    console.log(count);
+    return count;
+  };
+
+  const toggleLiked = (id) => {
+    const newMessages = [];
+    const messagesCopy = JSON.parse(JSON.stringify(messages));
+    for (const message of messagesCopy) {
+      if (message.id === id) {
+        message.liked = !message.liked;
+      }
+      newMessages.push(message);
+    }
+    setMessages(newMessages);
+    console.log('toggleLiked called');
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>
+          Chat between {chatMessages[0].sender} and {chatMessages[1].sender}
+        </h1>
+        <h2> {countLiked()} ❤️s</h2>
```

Comment:
Style: This should be a `section` or `p` or something since `h1`, `h2`, etc should be reserved for titles of parts of the page and this doesn't actually have content associated with it.